electrum/gui/qt/__init__.py

```diff
@@ -398,6 +403,72 @@ def start_new_window(
         return window
 
     def _start_wizard_to_select_or_create_wallet(self, path) -> Optional[Abstract_Wallet]:
+        wizard = QENewWalletWizard(self.config, self.app, self.plugins, self.daemon, path)
+        result = wizard.exec()
+        # TODO: use dialog.open() instead to avoid new event loop spawn?
+        self.logger.info(f'{result}')
+        if result == QENewWalletWizard.Rejected:
+            self.logger.info('ok bye bye')
+            return
+
+        d = wizard.get_wizard_data()
+
+        if d['wallet_is_open']:
+            for window in self.windows:
+                if window.wallet.storage.path == d['wallet_name']:
+                    return window.wallet
+            raise Exception('found by wizard but not here?!')
+
+        if not d['wallet_exists']:
+            self.logger.info('about to create wallet')
+            wizard.create_storage()
+            if d['wallet_type'] == '2fa' and 'x3/' not in d:
+                return
+            wallet_file = wizard.path
+        else:
+            wallet_file = d['wallet_name']
+
+        storage = WalletStorage(wallet_file)
+        if storage.is_encrypted_with_user_pw() or storage.is_encrypted_with_hw_device():
+            storage.decrypt(d['password'])
+
+        db = WalletDB(storage.read(), storage=storage, manual_upgrades=True)
+        if db.requires_split() or db.requires_upgrade():
+            try:
+                wizard.run_upgrades(db)
+            except UserCancelled:
+                return
+
+        if action := db.get_action():
+            # wallet creation is not complete, 2fa online phase
+            assert action[1] == 'accept_terms_of_use', 'only support for resuming trustedcoin split setup'
+            k1 = load_keystore(db, 'x1/')
+            if 'password' in d and d['password']:
+                xprv = k1.get_master_private_key(d['password'])
+            else:
+                xprv = db.get('x1/')['xprv']
+            data = {
+                'wallet_name': os.path.basename(wallet_file),
+                'xprv1': xprv,
+                'xpub1': db.get('x1/')['xpub'],
+                'xpub2': db.get('x2/')['xpub'],
+            }
+            wizard = QENewWalletWizard(self.config, self.app, self.plugins, self.daemon, path,
+                                       start_viewstate=WizardViewState('trustedcoin_tos_email', data, {}))
+            result = wizard.exec()
+            if result == QENewWalletWizard.Rejected:
+                self.logger.info('ok bye bye')
+                return
+            db.put('x3/', wizard.get_wizard_data()['x3/'])
+            db.write()
+
+        wallet = Wallet(db, config=self.config)
+        wallet.start_network(self.daemon.network)
+        self.daemon.add_wallet(wallet)
+        return wallet
+
+
+    def _start_wizard_to_select_or_create_wallet2(self, path) -> Optional[Abstract_Wallet]:
```

Comment:
is this unused?

Reply:
Yes, I used this to quickly switch between old wizard and new wizard entry. I'll remove it.